src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  const currentState = { ...state };
+
+  for (const action of actions) {
+    const { extraData, type, keysToRemove } = action;
+
+    switch (type) {
+      case 'clear':
+        // eslint-disable-next-line no-const-assign
+        for (const key in currentState) {
+          delete currentState[key];
+        }
+        break;
+
+      case 'addProperties':
+        Object.assign(currentState, extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of keysToRemove) {
+          delete currentState[key];
```

Comment:
Again, deleting keys from 'currentState' is mutating the state object. You should create a new object for the 'removeProperties' action to avoid mutations.